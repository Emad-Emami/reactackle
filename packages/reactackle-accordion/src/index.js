import Accordion from './Accordion';
import AccordionItem from './AccordionItem/AccordionItem';

export { Accordion, AccordionItem };
