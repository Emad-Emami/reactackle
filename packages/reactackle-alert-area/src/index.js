import AlertArea from './AlertArea';

export { AlertArea };
