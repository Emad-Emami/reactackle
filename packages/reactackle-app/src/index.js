import App from './App';

export { App };
export * from './BottomRegion';
export * from './TopRegion';
export * from './MainRegion';
