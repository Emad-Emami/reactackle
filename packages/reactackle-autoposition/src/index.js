import AutoPosition, { recalculateAllAutoPosition } from './AutoPosition';

export { AutoPosition, recalculateAllAutoPosition };
