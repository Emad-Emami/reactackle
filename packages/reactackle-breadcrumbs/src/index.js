import Breadcrumbs from './Breadcrumbs';

export { Breadcrumbs };
export * from './BreadcrumbsItem';
