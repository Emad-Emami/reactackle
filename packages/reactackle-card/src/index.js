import Card from './Card';

export { Card };
export * from './CardActions';
export * from './CardAreaMain';
export * from './CardAreaSide';
export * from './CardContent';
export * from './CardHeaderPrimary';
export * from './CardHeaderSecondary';
export * from './CardMedia';
