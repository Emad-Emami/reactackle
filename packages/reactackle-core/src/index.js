// theme
// defaultTheme
// composeReactackleTheme
// defaultThemeBlueprintBroadcast
// registerDefaultComponentTheme ++
// extractThemeOrDefault ++

// styleHelpers - for combining local theme
// export everything

// styleFunctions
// export everything

// - resetStyles
// - globalStyles
// - heladerClasses

// base
// export nothing

// constants
// export everything

// mixins
// export everything

// import {
//   registerDefaultComponentTheme,
//   extractThemeOrDefault,
// } from './theme';

// export {
//   registerDefaultComponentTheme,
//   extractThemeOrDefault,
// };

export * from './theme';
export * from './mixins';
export * from './styleFunctions';
export * from './styleHelpers';
export * from './utils';
export * from './constants/constants';
export * from './hoc/withExternalProps';
export * from './globalStyles';
export * from './resetStyles';
export * from './Theme/Theme';
export * from './Theme/withTheme';
