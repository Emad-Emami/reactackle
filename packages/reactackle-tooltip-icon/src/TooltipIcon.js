'use strict';

import React from 'react';
import PropTypes from 'prop-types';
import { withTooltip } from 'reactackle-tooltip';
import { noop, registerDefaultComponentTheme } from 'reactackle-core';
import { TooltipIconStyled } from './styles/TooltipIconStyled';
import componentTheme from './styles/theme';

registerDefaultComponentTheme('tooltipIcon', componentTheme);

/**
 * Combined with tooltip
 */
const propTypes = {
  /**
   * Text of the Tooltip
   */
  text: PropTypes.string.isRequired,
  /**
   * @ignore
   */
  toggleTooltip: PropTypes.func,
  /**
   * @ignore
   */
  showTooltip: PropTypes.func,
  /**
   * @ignore
   */
  hideTooltip: PropTypes.func,
  /**
   * @ignore
   */
  Tooltip: PropTypes.func,
};

const defaultProps = {
  toggleTooltip: noop,
  showTooltip: noop,
  hideTooltip: noop,
  Tooltip: noop,
};

export const TooltipIconComponent = props => {
  const { Tooltip } = props;
  const wrapperProps = {
    onClick: props.toggleTooltip,
    onFocus: props.showTooltip,
    onBlur: props.hideTooltip,
    onMouseEnter: props.showTooltip,
    onMouseLeave: props.hideTooltip,
    tabIndex: 0,
  };
  return (
    <TooltipIconStyled {...wrapperProps}>
      <span>?</span>
      <Tooltip text={props.text} />
    </TooltipIconStyled>
  );
};

TooltipIconComponent.propTypes = propTypes;
TooltipIconComponent.defaultProps = defaultProps;
TooltipIconComponent.displayName = 'TooltipIcon';

export default withTooltip(TooltipIconComponent, true);
