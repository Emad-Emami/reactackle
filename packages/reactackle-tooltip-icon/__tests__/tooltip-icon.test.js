'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import { TooltipIcon } from '../src';

jest.mock('react-dom');

describe('<TooltipIcon/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <TooltipIcon text="test" />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
