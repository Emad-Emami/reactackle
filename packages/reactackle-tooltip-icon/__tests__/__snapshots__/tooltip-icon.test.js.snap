// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TooltipIcon/> renders correctly with default props 1`] = `
<div
  className="TooltipIconStyled-gADMPy UneHs"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  tabIndex={0}
>
  <span>
    ?
  </span>
</div>
`;
